packages/trace-viewer/src/sw/main.ts

```diff
@@ -95,6 +95,30 @@ async function doFetch(event: FetchEvent): Promise<Response> {
 
     const traceUrl = url.searchParams.get('trace');
 
+    if (relativePath.startsWith('/snapshot/')) {
+      const { snapshotServer } = loadedTraces.get(traceUrl!) || {};
+      if (!snapshotServer)
+        return new Response(null, { status: 404 });
+      const response = snapshotServer.serveSnapshot(relativePath, url.searchParams, url.href);
+      if (isDeployedAsHttps)
+        response.headers.set('Content-Security-Policy', 'upgrade-insecure-requests');
+      return response;
+    }
+
+    if (relativePath.startsWith('/closest-screenshot/')) {
+      const { snapshotServer } = loadedTraces.get(traceUrl!) || {};
+      if (!snapshotServer)
+        return new Response(null, { status: 404 });
+      return snapshotServer.serveClosestScreenshot(relativePath, url.searchParams);
+    }
+
+    if (!client) {
+      // vscode webview doesn't sent clientId under some circumstances
+      if (url.pathname.endsWith('embedded.html'))
```

Comment:
maybe a good time to remove the embedded.html changes? I this its all unused anyways.

Reply:
we discussed this on friday - we'll remove the changes, but I don't want that to block this PR. let's go ahead with this workaround, and remove it together with `embedded.html`